### ISSUE TITLE
Removing attributes of type 'asset' from scripts can't be applied to templates

### ISSUE THREAD

**Issue by MoserAlex** [playcanvas/editor#382]
When removing an attribute from a script on a template, the change can't be applied to the template if the attribute was a reference to an asset. It will always look as if the template has some changes compared to the original template.
You can revert the change to remove the blue highlight on the script/remove the notification at the top of the template instance temporarily. But once you reload the page or change to another scene with an instance of this template, the same changes are being shown again.

I created a small sample [project](https://playcanvas.com/editor/scene/1143142), I hope you can reproduce it there. The cube is a template instance which at one time held a reference to the plane template. I then removed the attribute from the script and now I can't apply this change to the template.

If you can't test it on my project, these are the steps I took:
* Create a template
* Add a script to the template
* Add an attribute to the script which holds an asset
* Apply the current state to the template
* Rename the attribute
* Now you should experience the same issue.

I tried debugging the editor a little bit, and it might be that the template conflict is missing an "override_type" for removing an attribute.
![template_override](https://user-images.githubusercontent.com/11185247/116884098-a2808b00-ac26-11eb-9c98-54044f538633.PNG)

**Comment by ctreffs** [comment]
We only found these override types in `editor.js`, of which none is applicable to the stated usecase:

```js
override_type = 'override_add_script'
override_type = 'override_delete_script'
override_type = 'override_reorder_children'
override_type = 'override_reorder_scripts'
override_type = 'override_reparent_entity'
```

**Comment by zachpeterpaul** [comment]
I've been working on some changes to how PlayCanvas handles script attributes, my changes will likely be deployed sometime next week and this issue should be fixed. I reproduced your problem in prod PlayCanvas, but it worked correctly for me locally. Conceptually, scripts are included in entities (and template assets) "by reference", so removing an attribute from a script should automatically remove it from all entities and template assets right after you click the 'parse' button. This is the change I have implemented.
